### PR TITLE
Dev/api checks and swagger update

### PR DESCRIPTION
**Pull request by tuz666**
Verified the following routes:

- pagination for latest Arcsi episodes at Home & Archive page (/item/latest)
- a more efficient one for Shows page (/show/list)

Updated swagger docs

### CODE REVIEW

**Review by baxgas — COMMENTED** [review]
@tuz666 
Thanks, I would suggest we keep this branch on `devarcsi` if possible and we can sync in the `ananasz`?
- For this endpoint `arcsi/show/schedule` didn't we agree to put in the latest episode from the show?
- The items list should give back also the show sub-array, eg. `arcsi/item/latest/?size=3&page=1` to have `"shows":[{"name":"Name", "id":1,  "archive_lahmastore_base_url":"show-slug", } ],` note the show-slug, which will be important for making the episode links. We don't necessarily use it, but I think the episode id could be also included in this endpoint. Basically, I think all the episode data (as in `arcsi/item/all` ) could be included, those few extra bytes don't matter so much, and at some point we might need some of them (eg. `airing`, `archived`, `language` ...)
- The same for `arcsi/show/merites/archive` endpoint: the item response should be full, as noted before 
- Still don't understand why there is no entry for the episode slug which is made from the play-file name? `"play_file_name":"slug.mp3"` is it so difficult to make it? You have some logic for it already which works at `/show/{show_slug}/episode/{episode_slug}`
- This one seems perfect as it is, thanks `/show/{show_slug}/page`
- I don't know if we agreed upon it in this iteration, but there should be some `item` search query for the archives page, so Gábor can make the filtering? eg. arcsi/item&search=episode

**Comment by tuz666** [comment]
@baxgas 
Nearly all of that you wrote are correct, most of the routes that you mentioned are still under development and not being used now.
In this PR I focused to fix the `/item/latest` and the `/show/list` routes. The first one was adjusted to return the `item`'s `shows` field, and the second one is optimized to return data for the Shows page.
I have a stashed commit for the search function, I'm going to open a PR for that one as well.

**Comment by gammaw** [comment]
> In this PR I focused to fix the /item/latest and the /show/list routes. The first one was adjusted to return the item's shows field, and the second one is optimized to return data for the Shows page.

Yeah, I can see these changes, you only changed the Schema definition for that. Looks good!

> I have a stashed commit for the search function, I'm going to open a PR for that one as well.

You mean in this PR? This commit seems to be it: https://github.com/mmmnmnm/lahmacun_arcsi/pull/143/commits/1c3afeaaf845e9827a37b3f2db3531442c9f17e1

After checking the `/item/latest` and `/show/list` and `/item/search` endpoints (could be only through inspecting examples from a browser), we could merge this PR, no @baxgas? Like we did in the previous case: have an arcsi master head which may be "ahead" of the current rework master head such that the additions in arcsi have been confirmed by the frontend developer, which would be you ;) 

One meta comment: @tuz666 did you intentionally keep ananasz in the API description? Well, it's still a good marker for unused APIs ;)

**Comment by baxgas** [comment]
@gammaw no-no please don't merge, there are still many issues that need to be solved. I know this is a huge work, but it doesn't make any sense to rush it half finished and broken (schedule is not fixed yet, latest pagination needs some over-thinking etc.) you can check the staging, it has the api more-less connected

**Comment by baxgas** [comment]
@gammaw @tuz666 also what I found is that since the endpoints give back already the paginated response (eg. `items size=12&page=1`) these should already filter out the not `active` or the episodes that are "later" than today, since if we make this filtering on frontend, we will have less than 12 (or whatever the number) per page. is it possible to make this in api?

**Comment by gammaw** [comment]
@tuz666 I changed the status of the PR to draft as we're working on `/show/{show_slug}/episode/{episode_slug}` on on `/show/schedule`, ok? 

For the `schedule2` API: @tuz666 I see that it would work and for sure it's better to have all items only server-side (and in memory) but still we deal with _all_ items in `shows_schema.dump(shows)`, which could sigificantly slow down loading the schedule page. I would try to understand why the query-based solution does not work. What do you think?

**Review by gammaw — COMMENTED** [review]
@tuz666 thanks for this bold change in the arcsi API! It looks good to me. I'll first review https://github.com/lahmacunradio/frontend/pull/252/files before I accept this PR, in case there would be any need for changing something (I hope there will be not). 

I added some comments, which are possible improvements for the future, but **I wouldn't change the code in this PR** so that we don't need to validate again, but feel free to comment back on them.

File: arcsi/api/show.py [hunk 1]
```diff
@@ -276,11 +318,14 @@ def view_show(id):
...
+        for item in serial_show["items"]:
+            item["name_slug"] = normalise(item["name"])
 
         return serial_show
     else:
         return make_response("Show not found", 404, headers)
 
+
 # We use this route on the legacy front-end show page
 @arcsi.route("show/<string:show_slug>/archive", methods=["GET"])
```
[the remark this answers is not in the document]
Comment (gammaw) [reply arcsi/api/show.py:330]: @tuz666 bingo, `/archive` is still in use indeed: https://github.com/lahmacunradio/frontend/blob/b1d50a4378b0881e1a7afc50e1376235478fece6/app/components/schedule/Fullitem.vue#L181

File: arcsi/api/show.py [hunk 2]
```diff
@@ -95,15 +91,60 @@ def make_show(self, data, **kwargs):
...
+@arcsi.route("/show/all_without_items", methods=["GET"])
+def list_shows_without_items():
+    return shows_schedule_schema.dumps(get_shows())   
+
 @arcsi.route("/show/schedule", methods=["GET"])
 def list_shows_for_schedule():
-    return shows_schedule_schema.dumps(get_shows())
+    do = DoArchive()
+    shows = Show.query.filter(Show.active == True).all()
+    for show in shows:
```
[the remark this answers is not in the document]
Comment (tuz666) [reply arcsi/api/show.py:103]: Nice find, that should be implemented in a future commit.

File: arcsi/api/show.py [hunk 3]
```diff
@@ -95,15 +91,60 @@ def make_show(self, data, **kwargs):
...
+                    item["name_slug"] = normalise(item["name"])
+                    show_json["items"] = item
+            # if there is no archived show return empty array
+            if (latest_item_found == False):
+                show_json["items"] = []
+    return json.dumps(shows_json)
+
 
 # We are gonna use this on the new page as the show/all
 @arcsi.route("/show/list", methods=["GET"])
```
[the remark this answers is not in the document]
Comment (tuz666) [reply arcsi/api/show.py:143]: I can't really get what you say here, both of them returns the shows without the items, the difference is that the first returns all available info except items while `/show/list` returns only those which are used on the Shows page

File: arcsi/api/show.py [hunk 4]
```diff
@@ -313,23 +359,60 @@ def view_show_archive(show_slug):
...
 def view_show_page(show_slug):
+    do = DoArchive()
     show_query = Show.query.filter_by(archive_lahmastore_base_url=show_slug)
     show = show_query.first()
     if show:
         # subquery = session.query(Item.id).filter(blabla -timedelta(day=1)).all().subquery()
         # query = session.query(Show).filter_by(blabla).(Item.id.in_(subquery))
-        show.items.filter(Item.play_date < datetime.today() - timedelta(days=1)).all()
-        return show_archive_schema.dump(show)
+        if show.cover_image_url:
```
[the remark this answers is not in the document]
Comment (tuz666) [reply arcsi/api/show.py:368]: Agree with you these modifications should be extracted to the utils method like the get_shows() method.

File: arcsi/api/show.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gammaw) [reply arcsi/api/show.py:143]: > both of them returns the shows without the items

That's why I think it's a bit counterintutive to call it `all_without_items` as the other also does not include the items.

[the remark this answers is not in the document]
Comment (tuz666) [reply arcsi/api/show.py:143]: The optimized one (`/show/list`) was in my original plans, there was a request by @baxgas to return not only those but every info that's why I created the `all_without_items`. It seems a bit redundant, but in the hope of that we could use it later, I thought we should keep it too.

[the remark this answers is not in the document]
Comment (gammaw) [reply arcsi/api/show.py:143]: Now I see. Thanks! Yeah, I can imagine that `/show/list` is not used at all at the moment (I didn't find it now after a quick search). Let's try to remember it and get rid of the one which is not used/needed when we do a rethink/refactor in the future.

[the remark this answers is not in the document]
Comment (gammaw) [reply arcsi/api/show.py:143]: https://github.com/lahmacunradio/frontend/issues/256 (frontend) and https://github.com/lahmacunradio/arcsi/issues/146 (arcsi)

File: arcsi/api/show.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gammaw) [reply arcsi/api/show.py:330]: https://github.com/lahmacunradio/frontend/issues/254

**Review by gammaw — APPROVED** [review]

**Pull request merged**